packages/plans-grid-next/src/hooks/data-store/use-segment-based-plans.ts

```diff
@@ -0,0 +1,51 @@
+import {
+	TYPE_BUSINESS,
+	TYPE_ECOMMERCE,
+	TYPE_ENTERPRISE_GRID_WPCOM,
+	TYPE_FREE,
+	TYPE_PERSONAL,
+	TYPE_PREMIUM,
+} from '@automattic/calypso-products';
+import useGuidedFlowGetSegment from './use-guided-flow-get-segment';
```

Comment:
I guess still WIP? This doesn't exist in the package, right❓ (I hope not, we'd need to refactor out now 😬 )

Reply:
Still WIP yes, the hook now lives in another part https://github.com/Automattic/wp-calypso/pull/91017/files. 

@agrullon95, @chriskmnds is right, as our hook to retrieve the segment is not correlated to the plans, but to be used in various places, so we need to find another place for it :)

Reply:
Maybe `packages/onboarding`?

Reply:
Maybe that's fine? 

We won't be querying/using that package in `plans-grid-next` either, so these still need to be pulled in differently here. The package is imported here, but I think it should be safe to remove - it's only for some styles that we may have cleaned up already.